src/SonarScanner.MSBuild.Common/RegexExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace SonarScanner.MSBuild.Common;
+
+public static class RegexExtensions
```

Comment:
What's this for? Is it due to T0004? That's a FP as we didn't think about a case where the class does not exist in the project.

If something times out in the scanner, it's fine to let it time out. It's not fine in sonar-dotnet.

See https://github.com/SonarSource/sonar-dotnet/issues/9328

If it's T0004, delete it

Reply:
It's due to T0004. Anyway, we are raising it both in the scanner and autoscan without providing the SafeRegex extension; I'll revert the changes here; it would make sense, IMO to have it on repositories where T0004 is enabled.